src/providers/krb5/krb5_child.c

```diff
@@ -1177,6 +1177,15 @@ static krb5_error_code validate_tgt(struct krb5_req *kr)
     } else {
         DEBUG(SSSDBG_CRIT_FAILURE ,"TGT failed verification using key " \
                                     "for [%s].\n", principal);
+        if (kerr == KRB5KDC_ERR_S_PRINCIPAL_UNKNOWN) {
+            DEBUG(SSSDBG_IMPORTANT_INFO,
+                  "The server principal was not found. If the user comes "
+                  "from a different domain than the host, please verify "
+                  "that Kerberos servers from all domains in the trust "
+                  "paths are reachable. Acquiring a ticket using the 'kvno' "
+                  "utility for the principal used for verification might "
+                  "also be used to debug the issue further\n");
+        }
         goto done;
```

Comment:
Hi Jakub and Sumit!

I do not like putting an url that can change into the debug message. I would either cut the message and leave just the half that I suggested in my previous comment or drop the patch completely as Jakub suggested.

Michal